src/order/upset.lean

```diff
@@ -0,0 +1,290 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Sara Rousta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Sara Rousta
+-/
+import data.set.lattice
+import data.set_like.basic
+
+/-!
+# Up-sets and down-sets
+
+This file defines upper and lower sets in an order.
+
+## Main declarations
+
+* `is_upset`: Predicate for a set to be an upper set.
+* `is_downset`: Predicate for a set to be a lower set.
```

Comment:
This explanation tells the user nothing - can you expand on what an upper set and lower set is?